src/app/games/lasca/LascaMove.ts

```diff
@@ -0,0 +1,141 @@
+import { Coord, CoordFailure } from 'src/app/jscaip/Coord';
+import { Move } from 'src/app/jscaip/Move';
+import { ArrayUtils } from 'src/app/utils/ArrayUtils';
+import { assert } from 'src/app/utils/assert';
+import { MoveEncoder } from 'src/app/utils/Encoder';
+import { Localized } from 'src/app/utils/LocaleUtils';
+import { MGPFallible } from 'src/app/utils/MGPFallible';
+import { MGPOptional } from 'src/app/utils/MGPOptional';
+import { MGPSet } from 'src/app/utils/MGPSet';
+import { JSONObject, JSONValue, JSONValueWithoutArray, Utils } from 'src/app/utils/utils';
+import { LascaState } from './LascaState';
+
+export class LascaMoveFailure {
+
+    public static readonly CAPTURE_STEPS_MUST_BE_DOUBLE_DIAGONAL: Localized = () => $localize`Capture must be double diagonal steps!`;
```

Comment:
Failure message must be user friendly!

Reply:
Advices should be more helpful!

Reply:
You must write failure message in a more user friendly way!

(You must capture by doing double diagonal steps)

Reply:
"A capture should be a double diagonal step. Look at the green indicators to help you!"
On a dit ça dans un autre contexte :p